nimx/button.nim

```diff
@@ -287,6 +287,7 @@ proc handleToggleTouchEv(b: Button, e: var Event): bool =
             b.value = toggleValue(b.value)
             b.sendAction(e)
         b.setState(if b.value == 1: bsDown else: bsUp)
+        b.setNeedsDisplay()        
```

Comment:
`setNeedsDisplay` is called from `setState` already. It should not be needed here.

Reply:
toggleValue has changed the value, which means the state already matches and setState does nothing even though the state has actually changed, hence the need for setNeedsDisplay()

Reply:
Ah I think I get it now. The `setNeedsDisplay` should follow `b.value = toggleValue(b.value)`. Am I correct?

Reply:
``toggleValue()`` seems to be only called in line 287.  
But if something else is likely to call toggleValue without setState, then should it be added to toggleValue instead?

Reply:
Meh toggleValue is merely a `not` operation. Changing `b.value` should be accompanied with `setNeedsDisplay`, but it's likely redisplay already happens for other reasons.

Reply:
PR has been updated to move the `setNeedsDisplay`